library/Icingadb/Hook/Common/TotalSlaReportUtils.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/* Icinga DB Web | (c) 2023 Icinga GmbH | GPLv2 */
+
+namespace Icinga\Module\Icingadb\Hook\Common;
+
+use Icinga\Module\Icingadb\ProvidedHook\Reporting\HostSlaReport;
+use Icinga\Module\Icingadb\Widget\EmptyState;
+use Icinga\Module\Reporting\Timerange;
+use ipl\Html\Html;
+
+use function ipl\I18n\t;
+
+trait TotalSlaReportUtils
+{
+    public function getHtml(Timerange $timerange, array $config = null)
+    {
+        $data = $this->getData($timerange, $config);
+
+        if (! count($data)) {
```

Comment:
You throw away the count of results here, although you need it again below anyway? Please only count once :wink: